### ISSUE TITLE
Constant VERSION is not set

### ISSUE THREAD

**Issue by ausi** [contao/installation-bundle#1]
The `VERSION` constant isn’t set in the install tool. If you use something like this in the DCA of an extension:

``` php
'sql' => version_compare(VERSION, '3.2', '<') ? "varchar(255) NOT NULL default ''" : "binary(16) NULL",
```

It installs the wrong column type.

**Comment by leofeyer** [comment]
It is not set because you are not supposed to use it anymore :)

**Comment by ausi** [comment]
OK, I must have missed this deprecation, sorry.
Is there an alternative to get the Contao core bundle version?
Or is it considered bad to check against the version?

**Comment by leofeyer** [comment]
You can use the `kernel.packages` parameter:

``` php
$packages = System::getContainer()->getParameter('kernel.packages');
$coreVersion = $packages['contao/core-bundle'];
```

However, I'm not sure if the parameter is set in the install tool. We may have to add it first.

**Comment by leofeyer** [comment]
@contao/developers What do you think?

**Comment by aschempp** [comment]
@leofeyer what's the question? I have no idea about the new install tool, never used it…

**Comment by leofeyer** [comment]
It is actually not about the install tool but the `VERSION` constant, which we have deprecated. @ausi is using it in his DCA files:

``` php
'sql' => version_compare(VERSION, '3.2', '<') ? "varchar(255) NOT NULL default ''" : "binary(16) NULL",
```

He now would have to use `kernel.packages` from the container, which makes things a lot more complicated, don't you think?

**Comment by aschempp** [comment]
it's always more complicated with a more sophisticated system ;-)
The solution for him would be very simple though: if `VERSION` is not defined, it's newer than 3.2.

``` php
'sql' => (defined('VERSION') && version_compare(VERSION, '3.2', '<')) ? "varchar(255) NOT NULL default ''" : "binary(16) NULL",
```

**Comment by leofeyer** [comment]
LOL

**Comment by ausi** [comment]
The problem I see is that it’s not consistent. The `VERSION` constant is defined if the DCA is loaded from the Contao backend, but not if it’s loaded from the install tool. And if you build the DCA cache via Contao first, the install tool would use it.

**Comment by aschempp** [comment]
But that would still be solved by my suggestion, right?

**Comment by leofeyer** [comment]
@ausi You are right, als long as we are maintaining the constants, they should be available in the install tool as well. I have fixed this in contao/standard-edition@64f998a855de98657b9d92f506f605c0d1d3b859.
